examples/frontend-re-frame/src/cljs/frontend_re_frame/core.cljs

```diff
@@ -0,0 +1,157 @@
+  (:require
+   [re-frame.core :as re-frame]
+   [reagent.core :as reagent]
+   [reitit.core :as r]
+   [reitit.coercion :as rc]
+   [reitit.coercion.spec :as rss]
+   [reitit.frontend :as rf]
+   [reitit.frontend.controllers :as rfc]
+   [reitit.frontend.easy :as rfe]))
+
+;;; Events ;;;
+
+(re-frame/reg-event-db
+ ::initialize-db
+ (fn [_ _]
+   {:current-route nil}))
+
+(re-frame/reg-event-fx
+ ::navigate
+ (fn [db [_ route]]
+   ;; See `navigate` effect in routes.cljs
+   {::navigate! route}))
+
+(re-frame/reg-event-db
+ ::navigated
+ (fn [db [_ new-match]]
+   (assoc db :current-route new-match)))
+
+;;; Subscriptions ;;;
+
+(re-frame/reg-sub
+ ::current-route
+ (fn [db]
+   (:current-route db)))
+
+;;; Views ;;;
+
+(defn home-page []
+  [:div
+   [:h1 "This is home page"]
+   [:button
+    ;; Dispatch navigate event that triggers a (side)effect.
+    {:on-click #(re-frame/dispatch [::navigate ::sub-page2])}
+    "Go to sub-page 2"]])
+
+(defn sub-page1 []
+  [:div
+   [:h1 "This is sub-page 1"]])
+
+(defn sub-page2 []
+  [:div
+   [:h1 "This is sub-page 2"]])
+
+;;; Effects ;;;
+
+;; Triggering navigation from events.
+(re-frame/reg-fx
+ ::navigate!
+ (fn [k params query]
+   (rfe/push-state k params query)))
+
+;;; Routes ;;;
+
+(defn href
+  "Return relative url for given route. Url can be used in HTML links."
+  ([k]
+   (href k nil nil))
+  ([k params]
+   (href k params nil))
+  ([k params query]
+   (rfe/href k params query)))
+
+(def routes
+  ["/"
+   [""
+    {:name      ::home
+     :view      home-page
+     :link-text "Home"
+     :controllers
+     [{;; Do whatever initialization needed for home page
+       ;; I.e (re-frame/dispatch [::events/load-something-with-ajax])
+       :start (fn [& params](js/console.log "Entering home page"))
+       ;; Teardown can be done here.
+       :stop  (fn [& params] (js/console.log "Leaving home page"))}]}]
+   ["sub-page1"
+    {:name      ::sub-page1
+     :view      sub-page1
+     :link-text "Sub page 1"
+     :controllers
+     [{:start (fn [& params] (js/console.log "Entering sub-page 1"))
+       :stop  (fn [& params] (js/console.log "Leaving sub-page 1"))}]}]
+   ["sub-page2"
+    {:name      ::sub-page2
+     :view      sub-page2
+     :link-text "Sub-page 2"
+     :controllers
+     [{:start (fn [& params] (js/console.log "Entering sub-page 2"))
+       :stop  (fn [& params] (js/console.log "Leaving sub-page 2"))}]}]])
+
+(defn on-navigate [new-match]
+  (let [old-match (re-frame/subscribe [::current-route])]
+    (when new-match
+      (let [cs (rfc/apply-controllers (:controllers @old-match) new-match)
+            m  (assoc new-match :controllers cs)]
+        (re-frame/dispatch [::navigated m])))))
+
+(def router
+  (rf/router
+   routes
+   {:data {:coercion rss/coercion}}))
+
+(defn init-routes! []
+  (js/console.log "initializing routes")
+  (rfe/start!
+   router
+   on-navigate
+   {:use-fragment true}))
+
+(defn nav [{:keys [router current-route]}]
+  (into
```

Comment:
It is best to use `(into [...] seq)` only when necessary, when items don't have keys and/or the seq is static. Here route name can be used as unique key, so you can just move the `for` loop inside `[:ul ]` and add `:key route-name` to `:li`.

Reply:
I seem to have a bad habit of overusing `into`. I changed it according to your comment.